worker/online_restore.go

```diff
@@ -594,3 +599,15 @@ func RunOfflineRestore(dir, location, backupId, keyFile string, key x.Sensitive,
 	// TODO: Fix this return value.
 	return LoadResult{Version: manifest.ValidReadTs()}
 }
+
+func addPredIntoRestorePreds(restorePreds []string, namespace int64) []string {
+	var newRestorePreds []string
```

Comment:
you could the the trick for filter in slice

## Filtering without allocating

This trick uses the fact that a slice shares the same backing array and capacity as the original, so the storage is reused for the filtered slice. Of course, the original contents are modified.

b := a[:0]
for _, x := range a {
	if f(x) {
		b = append(b, x)
	}
}